src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,45 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add country to DB: " + country, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        Optional<Country> country;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            country = Optional.ofNullable(session.get(Country.class, id));
+        } catch (Exception e) {
+            if (transaction != null) {
```

Comment:
Rollback in 'get' method is unnecessary as there are no changes to be rolled back in case of read operations.